library/src/main/java/org/apache/apex/malhar/lib/window/accumulation/Sort.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.lib.window.accumulation;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import org.apache.apex.malhar.lib.window.Accumulation;
+import org.apache.hadoop.classification.InterfaceStability;
+
+/**
+ * Sort accumulation
+ */
+@InterfaceStability.Evolving
+public class Sort<T> implements Accumulation<T, List<T>, List<T>>
+{
+  boolean sortDesc;
+  Comparator<T> comparator;
+  Comparator<T> descComparator;
+
+  /**
+   * @param sortDesc
+   *          true if descending sort order is required
+   */
+  public void setSortDesc(boolean sortDesc)
+  {
+    this.sortDesc = sortDesc;
+  }
+
+  public void setComparator(Comparator<T> comparator)
+  {
+    this.comparator = comparator;
+    this.descComparator = Collections.reverseOrder(comparator);
+  }
+
+  @Override
+  public List<T> defaultAccumulatedValue()
+  {
+    return new ArrayList<T>();
+  }
+
+  @Override
+  public List<T> accumulate(List<T> accumulatedValue, T input)
+  {
+    if (comparator == null) {
+      throw new RuntimeException("Comparator not provided, Tuple cannot be compared");
+    }
+    Comparator<T> accComparator = sortDesc ? descComparator : comparator;
+    //binarySearch returns location if input exists else returns (-(insertion point) - 1)
+    int index = Collections.binarySearch(accumulatedValue, input, accComparator);
```

Comment:
This will be suboptimal since the elements in the arraylist will have to be moved by 1 when inserting the element. I couldnot think of any predefined collections which could help us improve optimality. We can probably improve optimality by using SkipLists.